### PR TITLE
Support for variable digit lengths

### PR DESCRIPTION
**Pull request by SuperVirus**
Adding support for variable digits.
Battle.net requires 8 digits instead of 6.

### CODE REVIEW

File: app/src/main/java/org/shadowice/flocke/andotp/Database/Entry.java [hunk 1]
```diff
@@ -178,6 +203,10 @@ public boolean hasNonDefaultPeriod() {
         return this.period != TokenCalculator.TOTP_DEFAULT_PERIOD;
     }
 
+    public boolean hasNonDefaultDigits() {
+        return this.digits != TokenCalculator.TOTP_DEFAULT_DIGITS;
+    }
+
```
Comment (flocke) [inline app/src/main/java/org/shadowice/flocke/andotp/Database/Entry.java]: Not needed

File: app/src/main/java/org/shadowice/flocke/andotp/View/EntriesCardAdapter.java [hunk 2]
```diff
@@ -119,6 +119,12 @@ public void onBindViewHolder(EntryViewHolder entryViewHolder, int i) {
             entryViewHolder.hideCustomPeriod();
         }
 
+        if (entry.hasNonDefaultDigits()) {
+            entryViewHolder.showCustomDigits(entry.getDigits());
+        } else {
+            entryViewHolder.hideCustomDigits();
+        }
+
```
Comment (flocke) [inline app/src/main/java/org/shadowice/flocke/andotp/View/EntriesCardAdapter.java]: Not needed

File: app/src/main/java/org/shadowice/flocke/andotp/View/EntryViewHolder.java [hunk 3]
```diff
@@ -112,6 +116,15 @@ public void hideCustomPeriod() {
...
 
+    public void showCustomDigits(int digits) {
+        customDigitsLayout.setVisibility(View.VISIBLE);
+        customDigits.setText(String.format("%d", digits));
+    }
+
+    public void hideCustomDigits() {
+        customDigitsLayout.setVisibility(View.GONE);
+    }
+
```
Comment (flocke) [inline app/src/main/java/org/shadowice/flocke/andotp/View/EntryViewHolder.java]: Not needed

File: app/src/main/res/drawable/ic_digit6_gray.xml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (flocke) [inline app/src/main/res/drawable/ic_digit6_gray.xml]: Not needed

File: app/src/main/res/layout/component_card.xml [hunk 5]
```diff
@@ -110,6 +110,26 @@
...
+                    android:layout_height="match_parent"
+                    android:src="@drawable/ic_digit6_gray" />
+
+                <TextView
+                    android:id="@+id/customDigits"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:textStyle="bold" />
+            </LinearLayout>
+
```
Comment (flocke) [inline app/src/main/res/layout/component_card.xml]: Not needed

**Comment by SuperVirus** [comment]
I'll start a new pull request with the simplified version.

**Pull request closed without merging**